src/ValidationManifold.jl

```diff
@@ -1,52 +1,149 @@
+The `key` of the dictionary can be either a `Symbol` or a `Function`.
+For a `Function` the value is either a `Bool` to indicate that within this function no checks are performed
+or a vector of symbols to deactivate certain types of checks. Since these are semantic, they are capitalized..
+The same symbols can be used as a key in the dictionary to disable them for all functions.
+
+# Examples
+* `exp => [:Point]` excludes point checks in the `exp` function
+* `exp => [:Point, :Vector]` excludes point and vector checks in the `exp` function
+* `exp => true` excludes all checks in the `exp` function
+* `:Point => false` deactivates point checks in all functions
+
+Current Symbols are
+* `:Point`: checks for points
+* `:Vector`: checks for vectors
+* `:Output`: checks for output
+* `:Input`: checks for input variables
+
+This also means an input point can _either be left unchecked when not checking points,
+_or_ when not checking inputs.
 
 This manifold is a decorator for a manifold, i.e. it decorates a [`AbstractManifold`](@ref) `M`
 with types points, vectors, and covectors.
 
+# Fields
+
+* `manifold::M`: The manifold to be decorated
+* `mode::Symbol`: The mode to be used for error handling, either `:error` or `:warn`
+* `ignore::Dict{<:Union{Symbol, Function},Symbol}`: A dictionary of disabled checks
+
 # Constructor
 
-    ValidationManifold(M::AbstractManifold; error::Symbol = :error)
+    ValidationManifold(M::AbstractManifold; kwargs...)
+
+Generate the Validation manifold
 
-Generate the Validation manifold, where `error` is used as the symbol passed to all checks.
-This `:error`s by default but could also be set to `:warn` for example
+## Keyword arguments
+
+* `error::Symbol=:error`: specify how errors in the validation should be reported.
+  this is passed to [`is_point`](@ref) and [`is_vector`](@ref) as the `error` keyword argument.
+  Available values are `:error`, `:warn`, `:info`, and `:none`. Every other value is treated as `:none`.
+* `store_base_point::Bool=false`: specify whether or not to store the point `p` a tangent or cotangent vector
+  is associated with. This can be useful for debugging purposes.
+* `ignores=Dict{Union{Function,Symbol},Union{Symbol,Bool}()` a dictionary of disabled checks
 """
-struct ValidationManifold{𝔽,M<:AbstractManifold{𝔽}} <: AbstractDecoratorManifold{𝔽}
+struct ValidationManifold{
+    𝔽,
+    M<:AbstractManifold{𝔽},
+    D<:Dict{<:Union{Symbol,Function},Union{Bool,Vector{Symbol}}},
+} <: AbstractDecoratorManifold{𝔽}
     manifold::M
     mode::Symbol
+    store_base_point::Bool
+    ignore::D
 end
-function ValidationManifold(M::AbstractManifold; error::Symbol = :error)
-    return ValidationManifold(M, error)
+function ValidationManifold(
+    M::AbstractManifold;
+    error::Symbol = :error,
+    store_base_point::Bool = false,
+    ignore::D = Dict{Union{Symbol,Function},Union{Bool,Vector{Symbol}}}(),
```

Comment:
In which case can the value of `ignore` be a `Bool`? `_vMc` doesn't seem to handle that.

Reply:
This case https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/d5712ca9bbe5d6bf4e0e99c740fa0adc231802a9/src/ValidationManifold.jl#L93

The idea is: with `exp => [:Point]` you deactivate `is_points` in all exp calls – but if you want to deactivate _all_ point checks for all functions you do `:Point => false` (or since it is an ignore dict maybe also true instead, have to think about that; one could also ignore all points stuff if the key is present) cf https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/d5712ca9bbe5d6bf4e0e99c740fa0adc231802a9/src/ValidationManifold.jl#L21

Reply:
Hm, I'd suggest storing these two things separately for more clarity. Maybe `ignore_f` and `ignore_all`? Then `ignore_all` could also just be a `Vector{Symbol}` instead of `Dict` (you'd just check `(type in ignore_all) && return false`).

Reply:
the first could be `ignore_function` and the other `ignore_type`, sure, will do that.